tendrl/node_agent/manager/manager.py

```diff
@@ -331,15 +297,24 @@ def load_and_execute_sds_discovery_plugins(self):
 
 
 def main():
+    # Definitions
+    Tendrl.definitions = Tendrl.node_agent.objects.Definition()
+
+    # Register Config to tendrl.node_agent.objects namespace
+    Tendrl.config = Config()
+
     setup_logging(
-        config['log_cfg_path'],
-        config['log_level']
+        Tendrl.node_agent.objects.Config.data['log_cfg_path'],
+        Tendrl.node_agent.objects.Config.data['log_level']
     )
-# init definitions
-# init manager and object tree from etcd
-    machine_id = utils.get_machine_id()
 
-    m = NodeAgentManager(machine_id)
+    etcd_kwargs = {'port': config['etcd_port'],
+                   'host': config["etcd_connection"]}
+    Tendrl.etcd_orm = etcdobj.Server(etcd_kwargs=etcd_kwargs)
```

Comment:
These are some steps to mitigate that.
- All writes to go via the persister thread as it was intended since the start.
- Reads can happen anywhere in the code but via the etcd_orm.read()
- One can tinker with the etcd_client's "per_host_pool_size" and "lock_prefix" to get more perf
https://github.com/jplana/python-etcd/blob/master/src/etcd/client.py#L66

- As for the "watch", it shouldn have been done via the main etcd_client anyways. Watch doesnt block the actual etcd api, if you simply use a temporary etcd_client instance for all your watches, it should be ok